component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordResetEnforcer.java

```diff
@@ -346,8 +347,12 @@ private boolean isPasswordPolicyViolationError(Throwable e) {
         while (e != null) {
             if (e.getCause() instanceof PolicyViolationException) {
                 return true;
-            }
-            if (e.getCause() instanceof IdentityPasswordHistoryException) {
+            } else if (e.getCause() instanceof IdentityPasswordHistoryException) {
+                return true;
+            } else if (e.getCause() instanceof IdentityEventException &&
```

Comment:
can simplyfy as bellow

 else if (e.getCause() instanceof IdentityEventException && PasswordPolicyConstants.PASSWORD_HISTORY_VIOLATION_ERROR_CODE.equals(I(dentityEventException) e.getCause()).getErrorCode())) {

Reply:
fix in https://github.com/wso2-extensions/identity-outbound-auth-passwordPolicy/pull/30/commits/6051a95e02b102f6e5f40a8958a35e7f02b61805